lib/services/groups/groupService.js

```diff
@@ -172,20 +172,27 @@ function remove(service, subservice, resource, apikey, device, callback) {
         callback(null, deviceGroup._id);
     }
 
+    function unregisterDevice(device, cb) {
+        deviceService.unregister(device.id, service, subservice, function(error) {
+            if (error) {
+                cb(error);
+            }
+        });
+    }
+
     function deleteDevices(device, service, subservice, id, callback) {
         if(device) {
             deviceService.listDevices(service, subservice, function (error, devices) {
                 if (error) {
                    callback(error);
                 } else {
                     if (devices && devices.count > 0){
-                      for(var device of devices.devices) {
-                        deviceService.unregister(device.id, service, subservice, function(error) {
+
+                        async.map(devices.devices, unregisterDevice, function(error) {
                             if (error) {
                                 callback(error);
-                            }
+                            }                            
```

Comment:
Trailing whitespace should be removed?

Reply:
Fixed with 082030fb26574c2bb4f1992ced8cec24cd8d5e12